core/account-request-handler/src/web2_identity/mod.rs

```diff
@@ -0,0 +1,265 @@
+	client: RestClient<HttpClient<DefaultSend>>,
+	path: String,
+	query: Vec<(String, String)>,
+}
+
+impl<R> Web2IdentityVerification<R> {
+	fn make_client(&self) -> Result<Web2HttpsClient, Error> {
+		let request = &self.verification_request;
+		match request.validation_data {
+			Web2ValidationData::Twitter(TwitterValidationData { ref tweet_id }) => {
+				let token = std::env::var("TWITTER_AUTHORIZATION_TOKEN")
+					.map_err(|_| Error::OtherError("token error".to_string()))?;
+				let tweet_id = String::from_utf8(tweet_id.to_vec())
+					.map_err(|_| Error::OtherError("tweet_id utf8 error".to_string()))?;
+
+				let tweet_id = tweet_id.trim();
+				if tweet_id.is_empty() {
+					return Err(Error::OtherError("tweet_id is empty".to_string()))
+				}
+
+				let client = build_client_with_authorization(TWITTER_BASE_URL, token.as_str());
+
+				Ok(Web2HttpsClient {
+					client,
+					path: "/2/tweets".to_string(),
+					query: vec![
+						("ids".to_string(), tweet_id.to_string()),
+						("expansions".to_string(), "author_id".to_string()),
+					],
+				})
+			},
+			Web2ValidationData::Discord(ref validation_data) => {
+				let token = std::env::var("DISCORD_AUTHORIZATION_TOKEN")
+					.map_err(|_| Error::OtherError("token error".to_string()))?;
+				let channel_id = String::from_utf8(validation_data.channel_id.to_vec())
+					.map_err(|_| Error::OtherError("channel_id utf8 error".to_string()))?;
+				let message_id = String::from_utf8(validation_data.message_id.to_vec())
+					.map_err(|_| Error::OtherError("message_id utf8 error".to_string()))?;
+
+				let channel_id = channel_id.trim();
+				let message_id = message_id.trim();
+				if channel_id.is_empty() {
+					return Err(Error::OtherError("channel_id is empty".to_string()))
+				}
+				if message_id.is_empty() {
+					return Err(Error::OtherError("message_id is empty".to_string()))
+				}
+
+				let client = build_client_with_authorization(DISCORD_BASE_URL, token.as_str());
+				Ok(Web2HttpsClient {
+					client,
+					path: format!("/api/channels/{}/messages/{}", channel_id, message_id),
+					query: vec![],
+				})
+			},
+		}
+	}
+}
+
+impl<
+		A: AuthorApi<Hash, Hash>,
+		S: StfEnclaveSigning,
+		K: ShieldingCryptoDecrypt + ShieldingCryptoEncrypt + Clone,
+		R: UserInfo + DecryptionVerificationPayload<K> + Debug + DeserializeOwned + RestPath<String>,
+	> RequestHandler<K, A, S> for Web2IdentityVerification<R>
+{
+	type Response = R;
+
+	fn send_request(&self, request_context: &RequestContext<K, A, S>) -> Result<(), Error> {
+		let mut client = self.make_client()?;
+		let query: Vec<(&str, &str)> =
+			client.query.iter().map(|(k, v)| (k.as_str(), v.as_str())).collect();
+		let response: Self::Response = client
+			.client
+			.get_with::<String, R>(client.path, query.as_slice())
+			.map_err(|e| Error::RquestError(format!("{:?}", e)))?;
+		//TODO log level
+		log::warn!("response:{:?}", response);
+		self.handle_response(request_context, response)
+	}
+
+	fn handle_response(
```

Comment:
I think a trait bound to `Response` would be better, or just extend the `DecryptionVerificationPayload` trait